File: CarePackage/Software/Dropbox.cs
namespace CarePackage.Software;

public class Dropbox : BaseSoftware
{
    public override string           Key            { get; set; } = "dropbox";
    public override string           Name           { get; set; } = "Dropbox";
    public override SoftwareCategory Category       { get; set; } = SoftwareCategory.Utility;
    public override string           DownloadName   { get; set; } = "DropboxInstaller.exe";
    public override bool             IsArchive      { get; set; } = false;
    public override bool             ShouldCacheUrl { get; set; } = false;
    public override bool             RequiresAdmin  { get; set; } = false;
    public override Bitmap           Icon           { get; set; } = Resources.Icons.dropbox;
    public override string           Homepage       { get; set; } = "https://dropbox.com";

    public override Task<string> GetDownloadUrlAsync(CancellationToken ct)
        => Task.FromResult("https://www.dropbox.com/download?os=win&plat=win");
}

File: packages/app/src/services/software/definitions/dropbox.ts
import { ok } from 'neverthrow';
import { SoftwareCategory } from 'shared';
import type { ISoftwareDefinition } from 'shared';

export default class implements ISoftwareDefinition {
	public key = 'dropbox';
	public name = 'Dropbox';
	public category = [SoftwareCategory.FileManagement, SoftwareCategory.Utility];
	public downloadName = 'DropoboxInstaller.exe';
	public icon = 'dropbox.png';
	public homepage = 'https://dropbox.com';

	public async resolveDownloadUrl() {
		return ok('https://www.dropbox.com/download?os=win&plat=win');
	}
}
